src/packages/client/src/__tests__/integration/happy/exit-hook/test.ts

```diff
@@ -1,9 +1,6 @@
 import { getTestClient } from '../../../../utils/getTestClient'
 
 test('exit-hook', async () => {
```

Comment:
This test is super specific - it only tests that this is called when `SIGINT` is emitted. But isn't that only part of the use case?

Should we not also add a test case where we just run `$disconnect()` and then the `beforeExit` should be executed as well?

Reply:
Tackling via https://github.com/prisma/prisma/pull/7789 to figure out if this is _actually_ the case for NodeEngine that `$disconnect` also triggers this.

Reply:
As you found out, the Node Engine does not trigger `beforeExit` on `$disconnect`

Reply:
Indeed. I am sad.